posts/Transient-Cluster-on-AWS.md

```diff
@@ -0,0 +1,341 @@
+
+```python
+Steps=[
+        {
+            'Name': 'spark-submit',
+            'ActionOnFailure': 'TERMINATE_CLUSTER',
+            'HadoopJarStep': {
+                'Jar': 's3://sparkuser/jar/example.jar',
+                'Args': [
+                	'spark-submit','--executor-memory', '1g', '--driver-memory', '500mb',
+                	'--class', 'spark.exercise_1'
+                    
+                ]
+            }
+        }
+    ]
+```
+
+#Step 4 : Cloudwatch
+
+Create a rule in CloudWatch using following command. Mention the name of the rule(Ex: DailyLambdaFunctin) and schedule time(Ex: 0 10 * * ? *  Run at 10.00 am UTC every day ).
+
+```bash
+aws events put-rule --name "DailyLambdaFunction" --schedule expression "cron(0 10 * * ? *)"
+```
+
+In the Targets section under the Lambda function select the name of the Lambda function which needs to be triggered by CloudWatch. In our case, it is ‘Emr_Spark’ as shown below.
+
+(https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/Cloudwatch.png)
+
+#Step 5: SNS and S3
+
+A topic is created in SNS and subscriptions(Email address) are added with a message to the topic.
+
+```bash
+aws sns create-topic --name Emr_Spark
+```
+
+The access policy for SNS topic to access S3 bucket:
+
+```json
+{
+  "Version": "2008-10-17",
+  "Id": "__default_policy_ID",
+  "Statement": [
+    {
+      "Sid": "s3-event-notifier",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "s3.amazonaws.com"
+      },
+      "Action": "SNS:Publish",
+      "Resource": "arn:aws:sns:us-east-1:Topic_name",
+      "Condition": {
+        "ArnLike": {
+          "AWS:SourceArn": "arn:aws:s3:*:*:bucket_name"
+        }
+      }
+    }
+  ]
+}
+```
+
+(https://raw.githubusercontent.com/ippontech/blog-usa/master/images/SNS.png)
+
+Create an s3 bucket(sparkuser) and add the event trigger for the SNS topic. As soon as the file or object lands in this s3 bucket, an SNS notification is sent to subscribers.
+
+```bash
+aws s3api put-bucket-notification-configuration --bucket sparkuser --notification-configuration file://notification.json
+```
+
+The file notification.json is a JSON document in the current folder that specifies an SNS topic and an event type to monitor.
+
+```json
+{
+    "TopicConfigurations": [
+        {
+            "TopicArn": "arn:aws:sns:us-east-1:s3-Topic_name",
+            "Events": [
+                "s3:ObjectCreated:*"
+            ]
+        }
+    ]
+}
+```
+
+#Conclusion
+
+The Design used here is a cost-effective, automated and simple solution to run Spark jobs on a transient EMR Cluster. In the next iteration, we can improve the design by adding more detail log analysis in case of cluster failures, add more Lambda functions and make use of step-functions to coordinate these Lambda functions.
```

Comment:
```suggestion
The Design used here is a cost-effective, automated, and simple solution to run Spark jobs on a transient EMR Cluster. In the next iteration, we can improve the design by adding more detailed log analysis in case of cluster failures, add more Lambda functions, and make use of step-functions to coordinate these Lambda functions.
```